### PR TITLE
Fix: ci issues and cherry-pick from stable branch.

### LINKED ISSUE #2111: python-flake8 max-line-length failure
The setup.cfg defines a max-line-length = 120 and this is overwritten in the python-flake8 with 127.

The extra definition in the worker script should be removed so that the settings from setup.cfg are used

### PR DESCRIPTION
**Pull request by rohit2p**
**Purpose of PR?**:
This pull request is a follow-up to PR #2134 , which addressed the flake8 max line failure. in this PR i am trying to get the code in the develop as well by cherry-pick the previous one and also made some changes in `tests/_test_msui/test_mscolab_merge_waypoints.py` so the ci test will run successfully. 

Fixes #2111 

**Checklist:**
- [X] Bug fix. Fixes #2111

### CODE REVIEW

File: .github/workflows/python-flake8.yml [hunk 1]
```diff
@@ -30,5 +30,5 @@ jobs:
     - name: Lint with flake8
       run: |
         python -m pip install --upgrade pip
-        pip install flake8 flake8-builtins
```
Comment (matrss) [inline .github/workflows/python-flake8.yml:33]: This line should stay as it is, the flake8-builtins package adds additional checks that we want to have.

File: tests/_test_msui/test_mscolab_merge_waypoints.py [hunk 2]
```diff
@@ -132,7 +132,9 @@ def test_save_overwrite_to_server(self, mockbox):
         assert wp_server_before.lat != wp_local_before.lat
 
         # trigger save to server action from server options combobox
-        with mock.patch("mslib.msui.mscolab.MscolabMergeWaypointsDialog", AutoClickOverwriteMscolabMergeWaypointsDialog):
+        with mock.patch(
+            "mslib.msui.mscolab.MscolabMergeWaypointsDialog",
+                AutoClickOverwriteMscolabMergeWaypointsDialog):
```
Comment (matrss) [inline tests/_test_msui/test_mscolab_merge_waypoints.py:137]: I think arguments to the same function should be aligned with the same indentation, i.e. the A should be under the double quote. This is usually done to make them more easily readable at a glance.

**Review by matrss — CHANGES_REQUESTED** [review]
Thanks for the PR! I left some comments. Also, the PR contains the cherry-picked commit twice, one of which is empty. I think that should be deduplicated.

File: tests/_test_msui/test_mscolab_merge_waypoints.py (hunk 2 above)
Comment (rohit2p) [reply tests/_test_msui/test_mscolab_merge_waypoints.py:137]: yea but when i do that i get this error `.\tests\_test_msui\test_mscolab_merge_waypoints.py:137:13: E129 visually indented line with same indent as next logical line  `

File: .github/workflows/python-flake8.yml (hunk 1 above)
Comment (rohit2p) [reply .github/workflows/python-flake8.yml:33]: ok I'll do that

**Comment by rohit2p** [comment]
sorry for the late response i was trying to figure out on my own but i tried deduplicating this by using revert, rebase everything i can but but i cant seem to do it. could to help me out a little bit here

File: tests/_test_msui/test_mscolab_merge_waypoints.py (hunk 2 above)
Comment (matrss) [reply tests/_test_msui/test_mscolab_merge_waypoints.py:137]: You are right, I didn't think of that. We can leave it as it is then.

**Comment by matrss** [comment]
> sorry for the late response i was trying to figure out on my own but i tried deduplicating this by using revert, rebase everything i can but but i cant seem to do it. could to help me out a little bit here

The command you are looking for is `git reset HEAD~1`. What this will do is set HEAD, i.e. the latest commit of your branch, to be the commit right before your current HEAD, effectively removing the latest commit.

Alternatively you could also do an interactive rebase onto develop: `git rebase -i develop`. You should then see an editor in which you can change the command for the empty commit (last line) to be drop instead of pick. After saving and closing in the editor git should apply the rebase, dropping the empty commit.

I hope this helps!

File: .github/workflows/python-flake8.yml [hunk 3]
```diff
@@ -30,5 +30,5 @@ jobs:
...
-        flake8 --count --max-line-length=127 --statistics mslib tests
+        pip install flake8 flake8-builtin
```
Comment (matrss) [inline .github/workflows/python-flake8.yml]:
```suggestion
        pip install flake8 flake8-builtins
```

The package name is missing an "s" at the end. You can also see that in the failing CI runs output: https://github.com/Open-MSS/MSS/actions/runs/7629972323/job/20784618748?pr=2163.

**Comment by rohit2p** [comment]
Thankyou for you help i think i have dropped the empty one.
now, It seems like there is an issue with the` flake8-builtin` package. maybe if we specify a specific version of it like this ` flake8-builtin==<version>` this may fix this

File: .github/workflows/python-flake8.yml (hunk 3 above)
Comment (rohit2p) [reply .github/workflows/python-flake8.yml]: i am so sorry i missed that my bad

**Review by matrss — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**